files/max-pods-calculator.sh

```diff
@@ -144,8 +150,15 @@ fi
 MAX_POD_CEILING_FOR_LOW_CPU=110
 MAX_POD_CEILING_FOR_HIGH_CPU=250
 CPU_COUNT=$(echo $DESCRIBE_INSTANCES_RESULT | jq -r '.CpuCount' )
+
+if [ "$SHOW_MAX_ALLOWED" = true ] ; then
+  echo $max_pods
```

Comment:
Some kind of ceiling is better than nothing at all. With IP limitations out of the way, the theoretical number (which is what `$max_pods` is here) could be really high. A c5.xlarge could give you > 500 pods so I'm not sure how useful this might be.

I think https://github.com/awslabs/amazon-eks-ami/pull/759/files is probably a better approach where you can override the ceiling value to be used in case you want to go higher *or* lower than the 110/250 recommendation this script currently gives out. That way the caller of the script can explicitly guide what number they're comfortable with.

Reply:
Hi @suket22,

I agree with you on setting some ceiling. But sometimes I only need to calculate the exact number of pods that I could run in an specific EC2 instance size, how to set the right ceiling depends on multiple factors like the kind of workload, max number of pods allowed, etc.

In my opinion both #759 and this feature are useful.

Reply:
I'm revisiting some of these changes on the EKS-optimized AMI and I think this is valuable. I also realized that even outside of CNI Prefix Delegation, this script is currently spitting out 250 if you're using saying an m5.16xlarge.

Reply:
Any update?